src/Flowdock/RocketeerFlowdockMessage.php

```
@@ -0,0 +1,118 @@
+    private $externalThreadID = NULL;
+
+    /**
+     * RocketeerFlowdockMessage constructor
+     *
+     * @param string $flowToken Your source_token provided after adding the RocketeerFlowdock integration
+     * @param string @external_thread_id The ID to pass to flowdock to allow for clustered notifications
+     * @param ClientInterface $client HTTP Client Interface
+     */
+    public function __construct($flowToken, $externalThreadID, ClientInterface $client = NULL)
+    {
+        if ($client == NULL) {
+            $client = new Client();
+        }
+
+        $this->flowToken = $flowToken;
+        $this->client = $client;
+        $this->externalThreadID = $externalThreadID;
+    }
+
+    /**
+     * Notifies the Flowdock channel of a deployment stage being initiated
+     *
+     * @param Closure @task
+     * @param string $threadBody
+     * @return bool
+     * @throws \Exception
+     */
+    public function notify($task, $threadBody = NULL)
+    {
+        if ($threadBody == NULL) {
+            $threadBody = "There is currently no message configured";
+        }
+
+        $body = json_encode([
+            'flow_token' => $this->flowToken,
+            'event' => 'activity',
+            'author' => [
+                'name' => get_current_user(),
+            ],
+            'title' => $this->formatThreadBody($task, $threadBody),
+            'external_thread_id' => $this->externalThreadID,
+            'thread' => [
+                'title' => $task->config->get('rocketeer-flowdock::thread_title'),
+                'body' => ''
+            ]
+        ]);
+
+        $headers = ['Content-Type' => 'application/json'];
+
+        $request = new Request('POST', self::MESSAGE_API, $headers, $body);
+        $response = $this->client->send($request);
+
+        if ($response->getStatusCode() != 202) {
+            throw new \Exception("Error: HTTP " . $response->getStatusCode() . " with message " . $response->getReasonPhrase());
```

Could you throw a more descriptive type of exception here rather than the catch-all Exception? Nothing wrong with what you have, but you might consider having some specific exception types so you can separate out generic "oh no something went wrong that was unexpected" exceptions from "oh this is a specific expected error response"